src/adapters/IDB.js

```diff
@@ -171,11 +171,28 @@ function createListRequest(cid, store, filters, done) {
 
   if (!indexField) {
     // Iterate on all records for this collection (ie. cid)
+    const isSubQuery = Object.keys(filters).some(key => key.includes(".")); // (ie. filters: {"article.title": "hello"})
+    if (isSubQuery) {
+      const newFilter = {};
+      Object.entries(filters).map(filter => {
+        filter.reduce((pv, cv) => {
+          if (pv.includes(".")) {
+            newFilter[pv.split(".")[1]] = cv;
+          } else {
+            newFilter[pv] = cv;
+          }
+        });
```

Comment:
I don't think the `reduce()` is used correctly here. It requires [an initial value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) and the function must return something.

Also, this implementation works with one level only, whereas ideally I think it should support filtering like `{ filters:  { "author.address.city": "barcelona" } }`